README.md

```diff
@@ -0,0 +1,453 @@
+
+- **refetch:** runs the fetch function every time the data is requested.
+- **expire:** when the data is requested, the fetch function is run only if data in the `Cache` is older than the expiration defined, otherwise the cached value is used.
+- **available:** when the data is requested, if a cached value is available it is always returned, otherwise the fetch function is run and the result stored in the `Cache` accordingly.
+
+All these utils ask you to specify custom [**`Setoid`**](https://github.com/gcanti/fp-ts/blob/master/docs/modules/Setoid.ts.md) instances that will be used to check if an input is already present in one of the `Chache`'s keys (if the check is successful `Avenger` will try to use that value, otherwise it will resort to the `Fetch` function) but you can (and should) use them together with one of the built-in implementations that automatically take care of it:
+- **queryShallow:** will use a `Setoid` instance that performs a shallow equality check to compare inputs.
+- **queryStrict:** will use a `Setoid` instance that performs a strict equality check to compare inputs.
+- **queryJSON:** will use a `Setoid` instance that performs a strict equality check after transforming the data via JSON stringification to compare inputs.
+
+Some examples will help clarify:
+```ts
+/*
+  this implementation will always re-run the `Fetch` function
+  even if valid cached data is already present
+  and use shallow equality to compare input
+*/
+const myQuery = queryShallow(fetchFunction, refetch);
+
+/*
+  this implementation will never run the `Fetch` function
+  unless no valid data is present in the Cache
+  and use strict equality to compare input
+*/
+const myQuery = queryStrict(fetchFunction, available);
+
+/*
+  this implementation will run the `Fetch` function only if no valid data is present in the Cache
+  or t > 10000 ms passed till the last time data was fetched
+  and use strict equality to compare input
+*/
+const myQuery = queryJSON(fetchFunction, expire(10000));
+```
+
+Each time the `Fetch` function is run with some `inputs`, those same `inputs` are used as a `key` to store the result obtained:
+```
+// usersCache is empty
+usersCache: {}
+
+//a user is fetched
+getUser({ userId: 1 }) -> { userName: "Mario" }
+
+// usersCache is now populated
+usersCache: {
+  [{ userId: 1 }]: { userName: Mario }
+}
+```
+
+From that moment onwards, when **Avenger** will need to decide if the data in our [**`Cache`**](#Cache) is present and still valid it will:
+
+1. attempt to retrieve data from the [**`Cache`**](#Cache)
+2. match the result against the cache strategy defined (for instance if we chose `refetch` the data will always be deemed invalid irrespective of the result).
+
+If the result is valid it is returned, otherwise the `Fetch` function will be re-run in order to get new and valid data. The two flows are relatively simple:
+##### Valid CacheValue
+!["cached flow"](docs/CachedValue.svg)
```

Comment:
![image](https://user-images.githubusercontent.com/6418684/60347026-bd47a000-99bc-11e9-93ed-6c70b6c9b49c.png)

Reply:
I don't get it in the readme it shows correctly.. 😓

Reply:
It could be just an error or the rich diff, ignore it 🤷‍♂